src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/CodedIndex.cs

```diff
@@ -6,38 +6,169 @@ namespace System.Reflection.Metadata.Ecma335
-        public static int ToMemberForwarded(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToMemberForwardedTag(handle.Kind));
-        public static int ToMemberRefParent(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToMemberRefParentTag(handle.Kind));
-        public static int ToMethodDefOrRef(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToMethodDefOrRefTag(handle.Kind));
-        public static int ToResolutionScope(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToResolutionScopeTag(handle.Kind));
-        public static int ToTypeDefOrRef(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToTypeDefOrRefTag(handle.Kind));
-        public static int ToTypeDefOrRefOrSpec(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToTypeDefOrRefOrSpecTag(handle.Kind));
-        public static int ToTypeOrMethodDef(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToTypeOrMethodDefTag(handle.Kind));
-        public static int ToHasCustomDebugInformation(EntityHandle handle) => MetadataTokens.GetRowNumber(handle).ToCodedIndex(ToHasCustomDebugInformationTag(handle.Kind));
-
-        private enum HasCustomAttribute
+        /// <summary>
+        /// Calculates a HasCustomAttribute coded index for the specified handle.
+        /// </summary>
+        /// <param name="handle">
+        /// <see cref="MethodDefinitionHandle"/>,
+        /// <see cref="FieldDefinitionHandle"/>,
+        /// <see cref="TypeReferenceHandle"/>,
+        /// <see cref="TypeDefinitionHandle"/>,
+        /// <see cref="ParameterHandle"/>,
+        /// <see cref="InterfaceImplementationHandle"/>,
+        /// <see cref="MemberReferenceHandle"/>,
+        /// <see cref="ModuleDefinitionHandle"/>,
+        /// <see cref="DeclarativeSecurityAttributeHandle"/>,
+        /// <see cref="PropertyDefinitionHandle"/>,
+        /// <see cref="EventDefinitionHandle"/>,
+        /// <see cref="StandaloneSignatureHandle"/>,
+        /// <see cref="ModuleReferenceHandle"/>,
+        /// <see cref="TypeSpecificationHandle"/>,
+        /// <see cref="AssemblyDefinitionHandle"/>,
+        /// <see cref="AssemblyReferenceHandle"/>,
+        /// <see cref="AssemblyFileHandle"/>,
+        /// <see cref="ExportedTypeHandle"/>,
+        /// <see cref="ManifestResourceHandle"/>,
+        /// <see cref="GenericParameterHandle"/>,
+        /// <see cref="GenericParameterConstraintHandle"/> or
+        /// <see cref="MethodSpecificationHandle"/>.
+        /// </param>
+        /// <exception cref="ArgumentException">Unexpected handle kind.</exception>
+        public static int HasCustomAttribute(EntityHandle handle) => (handle.RowId << (int)HasCustomAttributeTag.BitCount) | (int)ToHasCustomAttributeTag(handle.Kind);
```

Comment:
Nah, github is just badly designed and shouldn't be used as a measure of reasonable line length, imo. if I open the source file in VS and resize the window to see the same I see here I get very small window. Nobody has such a small monitor.